hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -508,18 +538,19 @@ public static String exportedSnapshotMapName(String name) {
         return JobRepository.EXPORTED_SNAPSHOTS_PREFIX + name;
     }
 
-    void clearSnapshotData(long jobId, int dataMapIndex) {
-        String mapName = snapshotDataMapName(jobId, dataMapIndex);
-        try {
-            instance.getMap(mapName).clear();
-            logFine(logger, "Cleared snapshot data map %s", mapName);
-        } catch (Exception logged) {
-            logger.warning("Cannot delete old snapshot data  " + idToString(jobId), logged);
-        }
+    private static DAG deserializeDag(MasterContext masterContext, JobRecord jobRecord) {
+        JetService jetService = masterContext.getJetService();
+        ClassLoader classLoader = jetService.getJobExecutionService()
+                .getClassLoader(jobRecord.getConfig(), jobRecord.getJobId());
+
+        SerializationService serializationService = masterContext.nodeEngine().getSerializationService();
+        return deserializeWithCustomClassLoader(serializationService, classLoader, jobRecord.getDag());
     }
 
-    void cacheValidationRecord(@Nonnull String snapshotName, @Nonnull SnapshotValidationRecord validationRecord) {
-        exportedSnapshotDetailsCache.set(snapshotName, validationRecord);
+    private static Set<String> ownedObservables(DAG dag) {
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(dag.iterator(), 0), false)
+                .map(vertex -> (String) vertex.getMetaSupplier().getTags().get(ObservableRepository.OWNED_OBSERVABLE))
+                .collect(Collectors.toSet());
```

Comment:
The returned set includes `null` if there is any vertex which is not an observable sink.